File: Java/app/src/main/java/com/example/xamarinpresentationjava/EditActivity.java
package com.example.xamarinpresentationjava;

import android.content.Intent;
import android.os.Bundle;
import android.widget.Button;
import android.widget.EditText;

import androidx.appcompat.app.AppCompatActivity;

import java.util.Objects;

public class EditActivity extends AppCompatActivity {
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_edit);
        Button submitButton = findViewById(R.id.submitButton);

        Intent intent = this.getIntent();
        String firstName = intent.getStringExtra("FirstName");
        String lastName = intent.getStringExtra("LastName");

        EditText firstNameEdit = findViewById(R.id.firstName);
        EditText lastNameEdit = findViewById(R.id.lastName);

        firstNameEdit.setText(firstName);
        lastNameEdit.setText(lastName);

        submitButton.setOnClickListener(b -> {
            String newFirstName = firstNameEdit.getText().toString();
            String newLastName = lastNameEdit.getText().toString();
            Intent result = new Intent();
            result.putExtra("Idx", intent.getIntExtra("Idx", -1));

            if(!Objects.equals(newFirstName, firstName) || !Objects.equals(newLastName, lastName)) {
                result.putExtra("FirstName", newFirstName);
                result.putExtra("LastName", newLastName);
            } else {
                result.putExtra("FirstName", "");
                result.putExtra("LastName", "");
            }

            setResult(RESULT_OK, result);
            finish();
        });
    }
}

File: C#/Xamarin Presentation/Classes/EditActivity.cs
using Android.App;
using Android.Content;
using Android.OS;
using Android.Widget;
using AndroidX.AppCompat.App;

namespace Xamarin_Presentation.Classes
{
    [Activity]
    public class EditActivity : AppCompatActivity
    {
        protected override void OnCreate(Bundle savedInstanceState)
        {
            base.OnCreate(savedInstanceState);
            SetContentView(Resource.Layout.activity_edit);
            Button submitButton = FindViewById(Resource.Id.submitButton) as Button;

            Intent intent = this.Intent;
            string firstName = intent.GetStringExtra("FirstName");
            string lastName = intent.GetStringExtra("LastName");

            EditText firstNameEdit = FindViewById(Resource.Id.firstName) as EditText;
            EditText lastNameEdit = FindViewById(Resource.Id.lastName) as EditText;

            firstNameEdit.Text = firstName;
            lastNameEdit.Text = lastName;

            submitButton.SetOnClickListener(b => {
                string newFirstName = firstNameEdit.Text;
                string newLastName = lastNameEdit.Text;
                Intent result = new Intent();
                result.PutExtra("Idx", intent.GetIntExtra("Idx", -1));

                if (newFirstName != firstName || newLastName != lastName)
                {
                    result.PutExtra("FirstName", newFirstName);
                    result.PutExtra("LastName", newLastName);
                }
                else
                {
                    result.PutExtra("FirstName", "");
                    result.PutExtra("LastName", "");
                }

                SetResult(Result.Ok, result);
                Finish();
            });
        }
    }
}